Vector to Closest Word

nikit-srivastava opened the pull request:
Implemented method to fetch the closest word entry for a given vector using Euclidean distance.

nikit-srivastava committed:
Vector to Closest Word Entry

Implemented method to fetch the closest word for a given vector using
Euclidean distance

nikit-srivastava committed:
Adding method level descriptions

nikit91 commented:
@MichaelRoeder Please Review

MichaelRoeder commented on src/main/java/org/aksw/word2vecrestful/word2vec/Word2VecModel.java:
```diff
@@ -11,9 +17,79 @@
+	  VectorDimension vectorDimension;
+	  //for each dimension
+	  for(int i=0;i<vectorSize;i++) {
+		  dimVals = new float[word2vec.size()];
```
You could optimize this by reusing the `dimVals` vector. You are creating a new array for every dimension and throw away the old array. That is not really necessary ;)

nikit-srivastava replied:
Good point :) It will be taken care of

MichaelRoeder commented on src/main/java/org/aksw/word2vecrestful/word2vec/Word2VecModel.java:
```diff
@@ -11,9 +17,79 @@
+   * @param minDist - minimum distance constraint
+   * @return squared euclidean distance between two vector or -1
+   */
+  private double getEucDistSqr(float[] arr1, float[] arr2, double minDist) {
```
shouldn't it be "squared Euclidian distance" instead of "Euclidian distance squared" ? So something like `getSqEucDist`? :thinking:

nikit-srivastava replied:
Yes, that sounds better :) I will change it

MichaelRoeder reviewed (approved):
Nice implementation. I only have two minor comments but it is already good :smiley:

nikit-srivastava committed:
Vector to Closest Word: Suggestions + compareTo fix

Implemented Michael's suggestions for the code.
Handling implemented for dimensions with same range.
